surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -44,15 +49,17 @@ TestsToRun apply(Configuration configuration) {
                 + "For details on how to configure it head over to http://bit.ly/st-config");
             return Collections.emptySet();
         }
-        final Set<TestSelection> selectedTests = new LinkedHashSet<>();
+        final Collection<TestSelection> selectedTests = new ArrayList<>();
         for (final String strategy : strategies) {
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
             selectedTests.addAll(plannerForStrategy.getTests());
         }
         logger.info("Applied strategies: %s", strategies);
         logger.info("Applied usage: [%s]", configuration.getMode().getName());
 
-        return selectedTests
+        final Collection<TestSelection> testSelections = filterMergeAndOrderTestSelection(selectedTests, strategies);
```

Comment:
I totally understand. but what's wrong with collection if we are using `LinkedHashSet` for collecting it. I believe it's also give you ordered collection